src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -162,12 +162,50 @@ static std::string getLogicalOpStr(const LogicalOperator logOp) {
 // SECTION PREFILTER EXPRESSION (BASE CLASS)
 //______________________________________________________________________________
 std::vector<BlockMetadata> PrefilterExpression::evaluate(
-    const std::vector<BlockMetadata>& input, size_t evaluationColumn) const {
+    std::vector<BlockMetadata>& input, size_t evaluationColumn,
+    bool stripIncompleteBlocks) const {
+  if (!stripIncompleteBlocks) {
+    return evaluateAndCheckImpl(input, evaluationColumn);
+  }
+
+  if (input.size() < 3) {
+    return input;
+  }
+  // Helper to check the given block for column consistency up to the evaluation
+  // column.
+  const auto checkBlockIsInconsistent =
+      [evaluationColumn](const BlockMetadata& block) {
+        return getMaskedTriple(block.firstTriple_, evaluationColumn) !=
+               getMaskedTriple(block.lastTriple_, evaluationColumn);
+      };
+  std::optional<BlockMetadata> firstBlock = std::nullopt;
+  std::optional<BlockMetadata> lastBlock = std::nullopt;
```

Comment:
Don't you already have this lambda inside the `evaluateAndCheckImpl`? If so, you shouldn't duplicate it, but define it (then without the caputre) outside the single functions.